pkg/activator/handler/healthz_handler_test.go

```diff
@@ -77,3 +78,49 @@ func TestHealthHandler(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkHealthHandler(b *testing.B) {
+	examples := []struct {
+		name    string
+		headers http.Header
+		check   func() error
+	}{{
+		name:    "forward non-kubelet request",
+		headers: mapToHeader(map[string]string{"User-Agent": "chromium/734.6.5"}),
+		check:   func() error { return nil },
+	}, {
+		name:    "kubelet probe success",
+		headers: mapToHeader(map[string]string{"User-Agent": "kube-probe/something"}),
+		check:   func() error { return nil },
+	}, {
+		name:    "kubelet probe failure",
+		headers: mapToHeader(map[string]string{"User-Agent": "kube-probe/something"}),
+		check:   func() error { return errors.New("not ready") },
+	}}
+
+	logger := ktesting.TestLogger(b)
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+	req := httptest.NewRequest(http.MethodGet, "http://example.com", nil)
+	for i, e := range examples {
+		handler := HealthHandler{HealthCheck: e.check, NextHandler: baseHandler, Logger: logger}
+		req.Header = e.headers
+		resp := httptest.NewRecorder()
+		b.Run(fmt.Sprintf("%d-sequential", i), func(b *testing.B) {
+			for j := 0; j < b.N; j++ {
+				handler.ServeHTTP(resp, req)
+			}
+		})
+
+		b.Run(fmt.Sprintf("%d-parallel", i), func(b *testing.B) {
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					// In order to avoid concurrent map writes
+					if e.check() != nil {
+						resp = httptest.NewRecorder()
```

Comment:
I think you can move this on top of the for loop and drop the extra check.

Reply:
In order to avoid `concurrent map writes` issue added check 

Because in third scenario
```
name:    "kubelet probe failure",
headers: mapToHeader(map[string]string{"User-Agent": "kube-probe/something"}),
check:   func() error { return errors.New("not ready") },
```

it will do some operation [here](https://github.com/knative/serving/blob/master/pkg/activator/handler/healthz_handler.go#L32-L34) 

and remaining 2 cases will not do anything with **w** and **r**